typescript/infra/src/utils/sealevel.ts

```diff
@@ -0,0 +1,287 @@
+  const diff = ((expectedBigInt - actualBigInt) * 10000n) / actualBigInt; // multiply by 10000 for 2 decimal places
+  const percentStr = (Number(diff) / 100).toFixed(2);
+  return diff >= 0n ? `+${percentStr}%` : `${percentStr}%`;
+}
+
+/**
+ * Gas Oracle Utility Functions
+ */
+
+/**
+ * Format RemoteGasData for display
+ */
+export function formatRemoteGasData(data: any): string {
+  // Scale exchange rate by 1e19 (TOKEN_EXCHANGE_RATE_SCALE for Sealevel)
+  const exchangeRate = Number(data.token_exchange_rate) / 1e19;
+  // Convert gas price to lamports (assuming it's in smallest unit)
+  const gasPriceLamports = Number(data.gas_price);
+
+  return `Exchange rate: ${exchangeRate.toFixed(10)}, Gas price: ${gasPriceLamports.toLocaleString()}, Decimals: ${data.token_decimals}`;
+}
+
+/**
+ * Serialize the difference between actual and expected gas oracle configs
+ *
+ * NOTE: The token_decimals field should represent the REMOTE chain's decimals, not the local chain's.
+ * If you see huge percentage differences, it may be because the on-chain config was incorrectly set with
+ * local decimals instead of remote decimals (e.g., solaxy has 6 decimals locally, but ethereum has 18).
+ */
+export function serializeGasOracleDifference(
+  actual: SealevelRemoteGasData,
+  expected: SealevelRemoteGasData,
+  calculatePercentDifference: (actual: bigint, expected: bigint) => string,
+): string {
+  const exchangeRateDiff = calculatePercentDifference(
+    actual.token_exchange_rate,
+    expected.token_exchange_rate,
+  );
+  const gasPriceDiff = calculatePercentDifference(
+    actual.gas_price,
+    expected.gas_price,
+  );
+
+  // Calculate product diff (exchange rate * gas price)
+  const actualProduct =
+    BigInt(actual.token_exchange_rate) * BigInt(actual.gas_price);
+  const expectedProduct =
+    BigInt(expected.token_exchange_rate) * BigInt(expected.gas_price);
+  const productDiff = calculatePercentDifference(
+    actualProduct,
+    expectedProduct,
+  );
+
+  const exchangeRate = Number(expected.token_exchange_rate) / 1e19;
+  const gasPriceLamports = Number(expected.gas_price);
+
+  return `Exchange rate: ${exchangeRate.toFixed(10)} (${exchangeRateDiff}), Gas price: ${gasPriceLamports.toLocaleString()} (${gasPriceDiff}), Product diff: ${productDiff}`;
+}
+
+export const DEFAULT_MAX_SEALEVEL_BATCH_SIZE = 10;
+
+/**
+ * Batch and send multiple transactions with instructions
+ *
+ * @param connection - Solana RPC connection
+ * @param items - Items to process in batches
+ * @param createInstruction - Function to create instruction from batch of items
+ * @param signerKeypair - Keypair to sign transactions
+ * @param chain - Chain name for logging and priority fees
+ * @param formatBatch - Function to format batch for logging
+ * @param operationName - Description of operation for logging
+ * @param maxBatchSize - Maximum items per transaction (default 10)
+ */
+export async function batchAndSendTransactions<T>(
```

Comment:
- I'd move it to the top to stay together with `batchAndSendTransaction`
- The order of arguments seems somewhat random, I can suggest (connection, chain, signerKeypair, operationName, items, createInstruction, formatBatch,...)